src/daemon/Display.cpp

```diff
@@ -164,6 +164,11 @@ namespace SDDM {
         }
         if (findSessionEntry(mainConfig.Wayland.SessionDir.get(), autologinSession)) {
             sessionType = Session::WaylandSession;
+            if (m_displayServerType == X11DisplayServerType) {
+                // If it's an X11 display server, we'll have an (idling) Xorg process where
+                // m_terminalId is right now, we need to find another one
+                m_terminalId = VirtualTerminal::setUpNewVt();
```

Comment:
How does the non-autologin case of wayland on rootfull Xorg work?

Reply:
Yes, this did not make a lot of sense. I've moved it down to startAuth which is where we used to deal with m_terminalId anyway.

I reproduced that it didn't work to log into wayland from root-x11 and that it works after this patch.